src/lib/number.jl

```diff
@@ -54,4 +54,9 @@ end
 @adjoint conj(x::Number) = conj(x), r̄ -> (conj(r̄),)
 @adjoint imag(x::Number) = imag(x), ī -> (real(ī)*im,)
 
+# Improve Array `conj`, `real` and `imag` Performance
+@adjoint real(x::AbstractArray) = real(x), r̄ -> (real(r̄),)
+@adjoint conj(x::AbstractArray) = conj(x), r̄ -> (conj(r̄),)
+@adjoint imag(x::AbstractArray) = imag(x), ī -> (real(ī)*im,)
```

Comment:
I would prefer this was fixed by adding the underlying rules for `broadcasted`, so that it also works when one does `conj.(x)` etc.

Reply:
Do you mean adding backward for the following interface?
```julia
@adjoint broadcasted(::typeof(conj), x) 
```

Reply:
Thanks, after adding this interface, `conj.` also works. The only problem is, here the rule is duplicated. Is it possible to reuse the backward rule defined for element-wise `conj`?

Reply:
Not yet unless we share things via DiffRules or add a macro for this. This is fine for now, thanks.

Reply:
So with the `broadcasted` definitions added, can we remove these without losing performance?